hypertrace-graphql-labels-schema-api/src/main/java/org/hypertrace/graphql/label/fetcher/LabelFetcher.java

```diff
@@ -16,20 +15,18 @@ public LabelFetcher() {
   }
 
   static final class LabelFetcherImpl implements DataFetcher<CompletableFuture<LabelResultSet>> {
-    private final LabelJoinerBuilder requestBuilder;
-    private final LabelDao labelDao;
+    private final EntityAndRuleJoinerBuilder requestBuilder;
 
     @Inject
-    LabelFetcherImpl(LabelJoinerBuilder requestBuilder, LabelDao labelDao) {
+    LabelFetcherImpl(EntityAndRuleJoinerBuilder requestBuilder) {
       this.requestBuilder = requestBuilder;
-      this.labelDao = labelDao;
     }
 
     @Override
     public CompletableFuture<LabelResultSet> get(DataFetchingEnvironment environment) {
       return this.requestBuilder
           .build(environment.getContext(), environment.getSelectionSet())
-          .flatMap(request -> request.joinLabelsWithEntities())
+          .flatMap(request -> request.joinLabelsWithEntitiesAndRules())
```

Comment:
I guess it doesn't really matter, but this is that inversion discussed elsewhere - the joiner is wrapping the dao rather than the dao wrapping the joiner. Since the DAO is an abstraction over where the data comes from, joiner in my mind is an implementation detail.